### PR TITLE
Run tests for type inference on CI

### PR DESCRIPTION
**Pull request by shinh**

### CODE REVIEW

File: chainer_compiler/elichika/typing/types.py [hunk 1]
```diff
@@ -349,14 +349,14 @@ def TyChainerVariable(dtype, ndim=None, shape=None):
 
 # ------------------------ TypeChecker internal types --------------------------
 
-counter = 0
+var_counter = 0
```
Comment (momohatt) [inline chainer_compiler/elichika/typing/types.py]: FYI: Actually we don't really need this counter variable to distinguish different TyVars and I'm just keeping it for `show()`. If you think it should be removed, please feel free to do so.

**Comment by momohatt** [comment]
I forgot to mention that we need to change these 2 lines too
https://github.com/pfnet-research/chainer-compiler/blob/f2d62b50d6d608699fdd03c6d6c50829b9d6fd15/chainer_compiler/elichika/testtools/type_checker_tools.py#L97-L98

File: chainer_compiler/elichika/typing/types.py (hunk 1 above)
Comment (shinh) [reply chainer_compiler/elichika/typing/types.py]: Yeah, I realized this is only for debugging. I personally like these kind of features so I'd like to keep this at least for a while. Let's remove this if it turns out this isn't not very useful for debugging.

**Comment by shinh** [comment]
As for two lines in `type_checker_tools.py`, I modified it locally to update the testcases but decided not to update them in this PR. I think we can/want to introduce a script which re-generates all testcases. Let me see.

**Pull request merged**